File: src/main/java/com/github/vootelerotov/intellij/chill/pill/transformer/ClassVisitorAdapter.java
package com.github.vootelerotov.intellij.chill.pill.transformer;

import org.objectweb.asm.*;

class ClassVisitorAdapter extends ClassVisitor {

  private final ClassVisitor delegate;

  public ClassVisitorAdapter(ClassVisitor delegate) {
    super(Opcodes.ASM9);
    this.delegate = delegate;
  }

  @Override
  public void visit(int version, int access, String name, String signature, String superName, String[] interfaces) {
    delegate.visit(version, access, name, signature, superName, interfaces);
  }

  @Override
  public void visitSource(String source, String debug) {
    delegate.visitSource(source, debug);
  }

  @Override
  public ModuleVisitor visitModule(String name, int access, String version) {
    return delegate.visitModule(name, access, version);
  }

  @Override
  public void visitNestHost(String nestHost) {
    delegate.visitNestHost(nestHost);
  }

  @Override
  public void visitOuterClass(String owner, String name, String descriptor) {
    delegate.visitOuterClass(owner, name, descriptor);
  }

  @Override
  public AnnotationVisitor visitAnnotation(String descriptor, boolean visible) {
    return delegate.visitAnnotation(descriptor, visible);
  }

  @Override
  public AnnotationVisitor visitTypeAnnotation(int typeRef, TypePath typePath, String descriptor, boolean visible) {
    return delegate.visitTypeAnnotation(typeRef, typePath, descriptor, visible);
  }

  @Override
  public void visitAttribute(Attribute attribute) {
    delegate.visitAttribute(attribute);
  }

  @Override
  public void visitNestMember(String nestMember) {
    delegate.visitNestMember(nestMember);
  }

  @Override
  public void visitPermittedSubclass(String permittedSubclass) {
    delegate.visitPermittedSubclass(permittedSubclass);
  }

  @Override
  public void visitInnerClass(String name, String outerName, String innerName, int access) {
    delegate.visitInnerClass(name, outerName, innerName, access);
  }

  @Override
  public RecordComponentVisitor visitRecordComponent(String name, String descriptor, String signature) {
    return delegate.visitRecordComponent(name, descriptor, signature);
  }

  @Override
  public FieldVisitor visitField(int access, String name, String descriptor, String signature, Object value) {
    return delegate.visitField(access, name, descriptor, signature, value);
  }

  @Override
  public MethodVisitor visitMethod(int access, String name, String descriptor, String signature, String[] exceptions) {
    return delegate.visitMethod(access, name, descriptor, signature, exceptions);
  }

  @Override
  public void visitEnd() {
    delegate.visitEnd();
  }
}

File: src/main/kotlin/com/github/vootelerotov/intellij/chill/pill/transformer/ClassVisitorAdapter.kt
package com.github.vootelerotov.intellij.chill.pill.transformer

import org.objectweb.asm.*

open class ClassVisitorAdapter(private val delegate: ClassVisitor) : ClassVisitor(Opcodes.ASM9) {

  override fun visit(
    version: Int,
    access: Int,
    name: String?,
    signature: String?,
    superName: String?,
    interfaces: Array<out String>?
  ) {
    delegate.visit(version, access, name, signature, superName, interfaces)
  }

  override fun visitSource(source: String?, debug: String?) {
    delegate.visitSource(source, debug)
  }

  override fun visitModule(name: String?, access: Int, version: String?): ModuleVisitor {
    return delegate.visitModule(name, access, version)
  }

  override fun visitNestHost(nestHost: String?) {
    delegate.visitNestHost(nestHost)
  }

  override fun visitOuterClass(owner: String?, name: String?, descriptor: String?) {
    delegate.visitOuterClass(owner, name, descriptor)
  }

  override fun visitAnnotation(descriptor: String?, visible: Boolean): AnnotationVisitor {
    return delegate.visitAnnotation(descriptor, visible)
  }

  override fun visitTypeAnnotation(
    typeRef: Int,
    typePath: TypePath?,
    descriptor: String?,
    visible: Boolean
  ): AnnotationVisitor {
    return delegate.visitTypeAnnotation(typeRef, typePath, descriptor, visible)
  }

  override fun visitAttribute(attribute: Attribute?) {
    delegate.visitAttribute(attribute)
  }

  override fun visitNestMember(nestMember: String?) {
    delegate.visitNestMember(nestMember)
  }

  override fun visitPermittedSubclass(permittedSubclass: String?) {
    delegate.visitPermittedSubclass(permittedSubclass)
  }

  override fun visitInnerClass(name: String?, outerName: String?, innerName: String?, access: Int) {
    delegate.visitInnerClass(name, outerName, innerName, access)
  }

  override fun visitRecordComponent(name: String?, descriptor: String?, signature: String?): RecordComponentVisitor {
    return delegate.visitRecordComponent(name, descriptor, signature)
  }

  override fun visitField(
    access: Int,
    name: String?,
    descriptor: String?,
    signature: String?,
    value: Any?
  ): FieldVisitor {
    return delegate.visitField(access, name, descriptor, signature, value)
  }

  override fun visitMethod(
    access: Int,
    name: String?,
    descriptor: String?,
    signature: String?,
    exceptions: Array<out String>?
  ): MethodVisitor {
    return delegate.visitMethod(access, name, descriptor, signature, exceptions)
  }

  override fun visitEnd() {
    delegate.visitEnd()
  }
}